packages/shared/lib/integrations/scripts/webhook/webhook.manager.ts

```diff
@@ -36,20 +37,19 @@ const internalNango: InternalNango = {
 
         return syncConfigsWithWebhooks;
     },
-    executeScriptForWebhooks: async (integration: ProviderConfig, body: any, webhookType: string, connectionIdentifier: string) => {
+    executeScriptForWebhooks: async (integration: ProviderConfig, body: any, webhookType: string, connectionIdentifier: string, propName?: string) => {
         const syncConfigsWithWebhooks = await internalNango.getWebhooks(integration.environment_id, integration.id as number);
 
         if (!syncConfigsWithWebhooks) {
             return;
         }
-
         const syncClient = await SyncClient.getInstance();
 
-        if (!body[connectionIdentifier]) {
+        if (!get(body, connectionIdentifier)) {
             return;
         }
 
-        const connection = await connectionService.findConnectionByConnectionConfigValue(connectionIdentifier, body[connectionIdentifier]);
+        const connection = await connectionService.findConnectionByConnectionConfigValue(propName || connectionIdentifier, get(body, connectionIdentifier));
```

Comment:
Should we move the find connection into the integration specific logic and pass the connection in the `executeScriptForWebhooks`. I am just worried that when we add more webhook integration we have to keep adding additional parameters to deal with the specificity of each intregrations?

Reply:
I don't disagree - but think that adjustment if needed should happen when adding in the next webhook if needed. Don't think that should block this PR from going in.